proto/workflow_execution.proto

```diff
@@ -8,58 +8,118 @@ import "google/api/annotations.proto";
 import "google/protobuf/struct.proto";
 import "task_execution.proto";
 
+//available workflow types supported by this WES
+enum WorkflowType {
+    UNKNOWN = 0;
+    CWL = 1;
+    WDL = 2;
+}
+
+//Return envelope for workflow listing
+message WorkflowListResponse {
+   repeated JobDesc workflows = 1;
+   string nextPageToken = 2;
+}
+
+//Small description of workflows, returned by server during listing
+message JobDesc {
+  //REQUIRED
+  string workflow_id  = 1;
+  //REQUIRED
+  ga4gh_task_exec.State state = 2;
+}
 
+//workflow request object
 message WorkflowRequest {
-  string workflow_url = 1;
-  google.protobuf.Struct input = 2;
+  //OPTIONAL
+  //user name of the workflow
+  string name = 1;
+  //OPTIONAL
+  //parameter for execution engine to define/store group information
+  string project_id = 2;
+  //OPTIONAL
+  //free text description of task
+  string description = 3;
```

Comment:
Right. 

I'd be ok w/ these things if they were optional for the *implementer* but that doesn't really work as smoothly as being optional for the implementation. 

I can totally see why these bits of info are useful in the larger picture, but I don't think APIs like this should be attaching loosely connected stuff because then invariably what will happen is that there'll be an expectation that they're used somehow. If there's really some sort of structure on those data to me it implies another API/service.

To give a slightly different example than your last statement, I could see going down a path where there's an expectation that an engine is storing this data and one can retrieve it. No. That can't be the case. That's something else.